testing/trino-benchto-benchmarks/src/test/java/io/trino/sql/planner/AbstractHiveCostBasedPlanTest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.planner;
+
+import io.trino.plugin.hive.RecordingMetastoreConfig;
+import io.trino.plugin.hive.TestingHiveConnectorFactory;
+import io.trino.plugin.hive.metastore.UnimplementedHiveMetastore;
+import io.trino.plugin.hive.metastore.recording.HiveMetastoreRecording;
+import io.trino.plugin.hive.metastore.recording.RecordingHiveMetastore;
+import io.trino.spi.connector.ConnectorFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.URL;
+import java.nio.file.Paths;
+import java.util.Arrays;
+
+import static com.google.common.collect.MoreCollectors.onlyElement;
+import static io.trino.plugin.hive.metastore.recording.TestRecordingHiveMetastore.createJsonCodec;
+
+public abstract class AbstractHiveCostBasedPlanTest
```

Comment:
Do we still need this abstract class since you've removed TPCH/TPC-DS connector based tests?

Reply:
Yes it just abstracts out the creation of recording metastore

Reply:
And the plan for AbstractCostBasedPlanTest is to keep it generic such that we can extend it for different connectors (maybe jdbc?) in future. That's why it has an abstract method `getConnectorFactory`.

Reply:
Still, I think this makes `createLocalQueryRunner` in `AbstractCostBasedPlanTest` a dead code.

> we can extend it for different connectors (maybe jdbc?) in future. That's why it has an abstract method getConnectorFactory.

I wouldn't worry about future cases too much. If it happens that we need it, we can always split it up.

Reply:
done